mm/mm_heap/mm.h

```diff
@@ -250,6 +250,7 @@ struct mm_heap_s
    * immdiately.
    */
 
+  spinlock_t mm_spinlock;
```

Comment:
Sorry, I have a question which is mutex can replace with spinlock?

Reply:
It isn't a replacement, both mutex and spinlock are used in mm_heap_s:

1. mutex is used to protect most shared resource since malloc/realloc can be only called in non-interrupt context
2. spinlock is only used to protect the free delay list since free may be called when the interrupt is disabled

Reply:
If  malloc/realloc only be called in non-interrupt context, it could be replace with spinlock.
Do you have any benchmark to measure the time to select using spinlock or mutex?
I just interest in why select mutex instead of spinlock.

mutex will call enter_critical_section and it comsumes a lot of time.

Reply:
> If malloc/realloc only be called in non-interrupt context, it could be replace with spinlock. Do you have any benchmark to measure the time to select using spinlock or mutex?

We saw 100x in no SMP case.

> I just interest in why select mutex instead of spinlock.
> 

spinlock disables the interrupt, so the real-time become worse.

> mutex will call enter_critical_section and it comsumes a lot of time.

Yes.

Reply:
Got it. Thanks for your reply.

Reply:
Got it. Thank for your reply.